### ISSUE TITLE
Verifying SSL Configurations: FAILED - can't run setup twice

### ISSUE THREAD

**Issue by gsabran** [zodern/meteor-up#205]
I'm trying to set up SSL. First time I run `mup setup` I get an error:

```
x Verifying SSL Configurations: FAILED

    -----------------------------------STDERR-----------------------------------
     complete
    d9e2907e9fa4: Verifying Checksum
    d9e2907e9fa4: Download complete
    4abf3e40edd9: Verifying Checksum
    4abf3e40edd9: Download complete
    3c907a5d5f8f: Verifying Checksum
    3c907a5d5f8f: Download complete
    8db38a632903: Verifying Checksum
    8db38a632903: Download complete
    963207107048: Verifying Checksum
    963207107048: Download complete
    03e1855d4f31: Verifying Checksum
    03e1855d4f31: Download complete
    03e1855d4f31: Pull complete
    a3ed95caeb02: Pull complete
    304ae4ee074b: Pull complete
    963207107048: Pull complete
    d9e2907e9fa4: Pull complete
    4abf3e40edd9: Pull complete
    3c907a5d5f8f: Pull complete
    8db38a632903: Pull complete
    Digest: sha256:573a1dd3e0e7829ad16e813d87e79948e9f2ef2cfa84d17c2c4c220b6e651b12
    Status: Downloaded newer image for meteorhacks/mup-frontend-server:latest
    nginx: [emerg] SSL_CTX_use_PrivateKey_file("/private.key") failed (SSL: error:0B080074:x509 certificate routines:X509_check_private_key:key values mismatch)
    nginx: configuration file /opt/nginx/conf/nginx.conf test failed
    -----------------------------------STDOUT-----------------------------------
    bdf9b92aa3697c77e745082bf6e8e7145e5442526793b04be6c980bb031ef245
    ----------------------------------------------------------------------------
```

There seems to be some issue with the certificate. I'll look into that. Second time I run `mup setup` (without changing anything else) I now get:

```
x Verifying SSL Configurations: FAILED

    -----------------------------------STDERR-----------------------------------
    docker: Error response from daemon: Conflict. The name "/pie-dummy-http-server-for-ssl-check" is already in use by container bdf9b92aa3697c77e745082bf6e8e7145e5442526793b04be6c980bb031ef245. You have to remove (or rename) that container to be able to reuse that name..
    See 'docker run --help'.
    nginx: [emerg] SSL_CTX_use_PrivateKey_file("/private.key") failed (SSL: error:0B080074:x509 certificate routines:X509_check_private_key:key values mismatch)
    nginx: configuration file /opt/nginx/conf/nginx.conf test failed
    -----------------------------------STDOUT-----------------------------------
    ----------------------------------------------------------------------------
```

So that seems to suggest that even if the first try was unsuccessful it left the server in an unclean state that prevents further recovery.

**Comment by madushan1000** [comment]
I'll look into this. thanks.

**Comment by gsabran** [comment]
In case that's helpful, for this issue or other having issue with SSL: the original error I was having was that my certificate file was the bundle of the intermediate certificates with the omission of the my own certificate ! I got confused because godaddy gave me two `.crt` files, one that was my own certificate and one that was all the intermediate certificates. I had to bundle them all together to get mup to work.

As a side note, mup was deploying ok if I only used my SSL certificate without the intermediate certificates, even if the resulting authentification is not right. It was working fine in some browsers but Cloudfront that enforces a stricter policy was not working properly. I dunno what's the best behavior that mup should have in that case.

**Comment by gsabran** [comment]
Actually, the duplicated container error happens even if the keys are correct, simply because `mup setup` is run twice.

**Comment by zeroasterisk** [comment]
I also had similar problems, was on my end w/ wrong certs... but I had to manually kill and rm the test container

**Comment by lfilho** [comment]
@zeroasterisk , @gsabran , @madushan1000 how can i know which certificate to use? I have the `*.pem` that letsencrypt generated me and right now i'm using them like so:

``` javascript
        ssl: {
            port: 443,
            crt: './fullchain.pem',
            key: './privkey.pem'
        }
```

I didn't join any files. Should I do anything different? Right now my `mup deploy` works but when accessing it in the browser it redirectis to https://localhost for some reason

**Comment by Vercryger** [comment]
Any progress/fix for this?

**Comment by gsabran** [comment]
@lfilho my guess is that you're not setting `meteor.env.ROOT_URL` to `https://your-website.com`

**Comment by zodern** [comment]
This will be fixed in 1.3.3.
